mlir/include/mlir/Dialect/Arith/IR/ArithBase.td

```diff
@@ -181,4 +181,38 @@ def Arith_RoundingModeAttr : I32EnumAttr<
   let cppNamespace = "::mlir::arith";
 }
 
+//===----------------------------------------------------------------------===//
+// Arith_DenormalMode
+//===----------------------------------------------------------------------===//
+
+// Denormal mode is applied on operands and results. For example, if denormal =
+// preserve_sign, operands and results will be flushed to sign preserving zero.
+// We do not distinguish between operands and results.
+
+// The default mode. Denormals are preserved and processed as defined 
+// by IEEE 754 rules.
+def Arith_DenormalModeIEEE          : I32BitEnumAttrCaseNone<"ieee">;
+
+// A mode where denormal numbers are flushed to zero, but the sign of the zero 
+// (+0 or -0) is preserved.
+def Arith_DenormalModePreserveSign  : I32BitEnumAttrCase<"preserve_sign", 1>;
+
+// A mode where all denormal numbers are flushed to positive zero (+0), 
+// ignoring the sign of the original number.
+def Arith_DenormalModePositiveZero  : I32BitEnumAttrCase<"positive_zero",  2>;
+
+def Arith_DenormalMode : I32BitEnumAttr<
```

Comment:
Do we need more than one bit set at a time? If not, this should be a plain `I32EnumAttr`.

Reply:
For some target you can have different modifiers for the operands and results (see this LLVM commit: https://github.com/llvm/llvm-project/commit/a3c814d23497bc71b8ed53c35f773366aff02922) but practically I don't see a lot of use cases in LLVM (see also commit message).

Reply:
Could we have a test like this?